demo-apps/sc-dtls-example-client/src/main/java/org/eclipse/californium/scandium/examples/ExampleDTLSClient.java

```diff
@@ -118,55 +119,72 @@ public void receiveData(RawData raw) {
 		}
 	}
 
-	private void start(InetSocketAddress peer) {
+	public long nbMessageReceived() {
+		return count.get();
+	}
+
+	private void start() {
 		try {
 			dtlsConnector.start();
-			dtlsConnector.send(new RawData("HELLO WORLD".getBytes(), peer));
 		} catch (IOException e) {
-			LOG.log(Level.SEVERE, "Cannot send message", e);
+			LOG.log(Level.SEVERE, "Cannot start connector", e);
 		}
 	}
 
-	private long waitReady(long timeout, TimeUnit unit) {
-		try {
-			latch.await(timeout, unit);
-			dtlsConnector.destroy();
-		} catch (InterruptedException e) {
-		}
-		return count.get();
+	private void startTest(InetSocketAddress peer) {
+		dtlsConnector.send(new RawData("HELLO WORLD".getBytes(), peer));
 	}
 
-	public static ExampleDTLSClient startTest(String[] args) throws InterruptedException {
-
-		ExampleDTLSClient client = new ExampleDTLSClient();
-		InetSocketAddress peer;
-		if (args.length == 3) {
-			peer = new InetSocketAddress(args[1], Integer.parseInt(args[2]));
-		} else {
-			peer = new InetSocketAddress(InetAddress.getLoopbackAddress(), DEFAULT_PORT);
-		}
-		client.start(peer);
-		return client;
+	private void stop() {
+		if (dtlsConnector.isRunning())
+			dtlsConnector.destroy();
 	}
 
 	public static void main(String[] args) throws InterruptedException {
-		long count = 0;
-		long nanos = System.nanoTime();
 		int max = 1;
 		if (0 < args.length) {
 			max = Integer.parseInt(args[0]);
 		}
-		List<ExampleDTLSClient> clients = new ArrayList<>();
+		latch = new CountDownLatch(max);
+		List<ExampleDTLSClient> clients = new ArrayList<>(max);
+
+		// Create & start clients
 		for (int index = 0; index < max; ++index) {
-			ExampleDTLSClient client = startTest(args);
+			ExampleDTLSClient client = new ExampleDTLSClient();
+			client.start();
 			clients.add(client);
 		}
+
+		// Get peer address
+		InetSocketAddress peer;
+		if (args.length == 3) {
+			peer = new InetSocketAddress(args[1], Integer.parseInt(args[2]));
+		} else {
+			peer = new InetSocketAddress(InetAddress.getLoopbackAddress(), DEFAULT_PORT);
+		}
+
+		// Start Test
+		long nanos = System.nanoTime();
 		for (ExampleDTLSClient client : clients) {
-			count += client.waitReady(100000, TimeUnit.SECONDS);
+			client.startTest(peer);
 		}
 
+		// Wait for it
+		latch.await(max * 50, TimeUnit.MILLISECONDS);
```

Comment:
If some clients get stale because of dropped messages, the measured times are more or less useless, if we wait for them too long. So currently, the the short times seems to have more  advantages than disadvantages.